contrib/go/src/python/pants/contrib/go/subsystems/go_import_meta_tag_reader.py

```diff
@@ -65,13 +62,13 @@ def get_imported_repo(self, import_path):
     if not page_data:
       return None
 
-    root, vcs, url = self.find_meta_tag(page_data.text)
-    if root and vcs and url:
-      # Check to make sure returned root is an exact match to the provided import path. If it is
-      # not then run a recursive check on the returned and return the values provided by that call.
-      if root == import_path:
-        return ImportedRepo(root, vcs, url)
-      elif import_path.startswith(root):
-        return self.get_imported_repo(root)
+    for (root, vcs, url) in self.find_meta_tags(page_data.text):
```

Comment:
Its probably worth commenting on punting the longest prefix match amongst multiple here.